File: solutions/java/lasagna/3/src/main/java/Lasagna.java
public class Lasagna {
    
    private static final int LASAGNA_COOK_MINUTES = 40;
    private static final int LASAGNA_LAYER_PREP_MINUTES = 2;
    
    public int expectedMinutesInOven() {
        return LASAGNA_COOK_MINUTES;
    }
    
    public int remainingMinutesInOven(int minutes) {
        return expectedMinutesInOven() - minutes;
    }
    
    public int preparationTimeInMinutes(int layers) {
        return LASAGNA_LAYER_PREP_MINUTES * layers;
    }
    
    public int totalTimeInMinutes(int layers, int minutes) {
        return preparationTimeInMinutes(layers) + minutes;
    }
}

File: solutions/r/lasagna/2/lasagna.R

expected_minutes_in_oven <- function() {
  60
}

remaining_time_in_minutes <- function(time) {
  expected_minutes_in_oven() - time
}
prep_time_in_minutes <- function(layers)  {
  layers * 2
}

elapsed_time_in_minutes <- function(layers, minutes) {
  prep_time_in_minutes(layers) + minutes
}